shared/editor/nodes/CodeFence.ts

```diff
@@ -91,6 +94,9 @@ export default class CodeFence extends Node {
         language: {
           default: DEFAULT_LANGUAGE,
         },
+        diagram: {
+          default: DEFAULT_DIAGRAM_TOGGLE_STATE,
+        },
```

Comment:
This works well as a first pass but it has a couple of disadvantages which we need to overcome:

1. The toggle state is synced between editors, so if you have two people in the same doc they will see the state switching – ideally we'd keep this local so that one person can be editing the diagram and the other is just viewing it, or editing elsewhere without the content jumping because of the toggle.
2. This new `diagram` attribute will get added to every code block and cause "edits" to happen to existing documents.

We probably need to keep the toggle inside local state instead, this will mean that the class is added based on a Node decoration that is controlled by that state. I realize that this is quite involved if not familiar with Prosemirror – If  everything else is taken care of I could make this translation.